### PR TITLE
[AN-5712][AN-5734] Login issues and tracking

### PR DESCRIPTION
**Pull request by JPonte**
#### APK
[Download build #9729](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9729/artifact/build/artifact/wire-dev-PR1234-9729.apk)
[Download build #9734](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9734/artifact/build/artifact/wire-dev-PR1234-9734.apk)
[Download build #9735](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9735/artifact/build/artifact/wire-dev-PR1234-9735.apk)
[Download build #9736](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9736/artifact/build/artifact/wire-dev-PR1234-9736.apk)
[Download build #9738](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9738/artifact/build/artifact/wire-dev-PR1234-9738.apk)
[Download build #9739](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9739/artifact/build/artifact/wire-dev-PR1234-9739.apk)
[Download build #9740](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9740/artifact/build/artifact/wire-dev-PR1234-9740.apk)
[Download build #9742](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9742/artifact/build/artifact/wire-dev-PR1234-9742.apk)
[Download build #9743](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9743/artifact/build/artifact/wire-dev-PR1234-9743.apk)
[Download build #9744](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9744/artifact/build/artifact/wire-dev-PR1234-9744.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/appentry/VerifyPhoneFragment.scala [hunk 1]
```diff
@@ -179,16 +183,24 @@ class VerifyPhoneFragment extends BaseFragment[VerifyPhoneFragment.Container] wi
...
-          if (getActivity == null) return
-          KeyboardUtils.showKeyboard(getActivity)
-          editTextCode.requestFocus
-          phoneConfirmationButton.setState(PhoneConfirmationButton.State.INVALID)
-        })
-      case _ =>
+    appEntryController.verifyPhone(editTextCode.getText.toString).map { result =>
+      ZMessaging.currentAccounts.getActiveAccount.collect { case Some(acc) => acc.regWaiting }.map { reg =>
+        tracking.onActivation(result, SignInMethod(if (reg) Register else Login, Phone))
+      }
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/appentry/VerifyPhoneFragment.scala]: You can probably make a small helper method out of this - even an inner method `trackConfirmCode` or something like that

**Comment by someoneEsle** [comment]
Bug
Please test PRs before adding the QA label -- registration.entered_name is not tracked

**Comment by someoneEsle** [comment]
See Shumeng's comment on the ticket, it contains all the bugs

File: app/src/main/scala/com/waz/zclient/appentry/VerifyPhoneFragment.scala [hunk 2]
```diff
@@ -153,22 +157,32 @@ class VerifyPhoneFragment extends BaseFragment[VerifyPhoneFragment.Container] wi
...
+      ZMessaging.currentAccounts.getActiveAccount.collect { case Some(acc) => acc.regWaiting }.map { reg =>
+        tracking.onRequestResendCode(result, SignInMethod(if (reg) Register else Login, Phone), isCall = true)
+      }
+      result match {
+        case Left(entryError) =>
+          getContainer.showError(entryError)
+          editTextCode.requestFocus
+        case _ =>
+          Toast.makeText(getActivity, getResources.getString(R.string.new_reg__code_resent__call), Toast.LENGTH_LONG).show()
+      }
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/appentry/VerifyPhoneFragment.scala]: A lot of common code here - I think only `isCall = true` differs. Maybe make a method here?

**Pull request merged**